Core/NWNXCore.cpp

```diff
@@ -275,6 +275,35 @@ void NWNXCore::InitialSetupPlugins()
     }
 }
 
+void NWNXCore::InitialSetupCommands()
+{
+    m_services->m_commands->RegisterCommand("set", [](std::string& s)
+    {
+        size_t equals = s.find('=');
+        if (equals != std::string::npos)
+        {
+            std::string varname = s.substr(0, equals);
+            std::string value = s.substr(equals+1, std::string::npos);
+            if (setenv(varname.c_str(), value.c_str(), 1) == 0)
```

Comment:
I think this is problematic. setenv is not thread safe. I think a safer design would be to extend the Config service so that you can change config settings that way.